src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -639,7 +639,9 @@ void ClassFileParser::parse_constant_pool(const ClassFileStream* const stream,
           if (Signature::is_method(sig)) {
             // Format check method name and signature
             verify_legal_method_name(name, CHECK);
-            verify_legal_method_signature(name, sig, CHECK);
+            // Pass no_name so method name doesn't affect check for legal signature.
+            const Symbol* const no_name = vmSymbols::type_name(); // place holder
+            verify_legal_method_signature(no_name, sig, CHECK);
```

Comment:
verify_legal_method_signature() has the potential to throw an exception.  Now it will have no name to include in that exception method concerning an illegal method signature.

Reply:
Thanks Lois for finding this issue.

Please review commit 3 that restores passing the method name to verify_legal_method_signature() for use in exception messages and adds a flag parameter indicating if verify_legal_method_signature() should check if the method name and signature are compatible.

The change also adds a test case to check that the correct method name is in the exception message when verify_legal_method_signature() finds a bad method signature.